src/librustdoc/doctest.rs

```diff
@@ -76,6 +76,17 @@ pub(crate) fn generate_args_file(file_path: &Path, options: &RustdocOptions) ->
         content.push(format!("-Z{unstable_option_str}"));
     }
 
+    for compilation_args in &options.doctest_compilation_args {
+        for flag in compilation_args
+            .split_whitespace()
+            .map(|flag| flag.trim())
+            .filter(|flag| !flag.is_empty())
+        {
+            // Very simple parsing implementation. Might be a good idea to correctly handle strings.
+            content.push(flag.to_string());
```

Comment:
This has been brought up a load of times, but I'm not sure this is the right thing to do here? Is there any prior art for passing flags down from a flag (and not an env-var).

(The good news is that this isn't insta-stable, so we can change the edge-behaviour later before stabilization)

Reply:
There's two correct options I know of:

 * Pass one flag at a time `--doctest-compilation-arg -Dwarnings --doctest-compilation-arg -Adead-code`
 * Pass the flags as a single string in a proper encoding, such as toml

Reply:
> Pass one flag at a time `--doctest-compilation-arg -Dwarnings --doctest-compilation-arg -Adead-code`.

You can currently do it with the option. :)

> Pass the flags as a single string in a proper encoding, such as toml

That could work too, but I'm not sure if it's something common enough to justify it here. Parsing shell quotes would be much simpler.

Reply:
> You can currently do it with the option. :)

No, the current implementation would mishandle trying to pass a space containing arg.

Reply:
Yes, I meant passing the same option flag multiple times.

Reply:
Code has evolved quite a lot since then. It's now how you would expect a shell environment to parse the command.